lib/src/warnings.dart

```diff
@@ -498,6 +501,13 @@ class PackageWarningCounter {
   UnmodifiableMapView<Element?, Map<PackageWarning, Set<String>>>
       get countedWarnings => UnmodifiableMapView(_countedWarnings);
 
+  /// An unmodifiable map view of all counted warnings related by their element,
+  /// warning type, and message.
+  UnmodifiableMapView<Element2?, Map<PackageWarning, Set<String>>>
+      get countedWarnings2 => UnmodifiableMapView(_countedWarnings.map((key,
+              value) =>
+          MapEntry(key != null ? (key.asElement2 as Element2) : null, value)));
```

Comment:
This is a strange cast `key.asElement2 as Element2`.
We expect to get `Element2` from `asElement2`, right?

Reply:
Missed that, fixed. Would attribute this to the auto complete AI plugin.